functionaltests/internal/kbclient/client_test.go

```diff
@@ -0,0 +1,150 @@
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package kbclient_test
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"os"
+	"path"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/dnaeon/go-vcr/cassette"
+	"github.com/dnaeon/go-vcr/recorder"
+	"github.com/itchyny/gojq"
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/apm-server/functionaltests/internal/kbclient"
+)
+
+// getHttpClient instantiate a http.Client backed by a recorder.Recorder to be used in testing
+// scenarios.
+// To update any fixture, delete it from the filesystem and run the test again.
+// NOTE: always remember to call (recorder.Recorder).Stop() in your test case.
+func getHttpClient(t *testing.T) (*recorder.Recorder, *http.Client) {
+	t.Helper()
+
+	var err error
+	var rec *recorder.Recorder
+	var hc *http.Client
+
+	tr := http.DefaultTransport
+	rec, err = recorder.NewAsMode(path.Join("testdata", "fixtures", t.Name()), recorder.ModeReplaying, tr)
+	require.NoError(t, err)
+
+	// Filter out dynamic & sensitive data/headers before saving the fixture.
+	// Tests will be using the real values when recording the interactions.
+	rec.AddSaveFilter(func(i *cassette.Interaction) error {
+		delete(i.Request.Headers, "Authorization")
+
+		var sanitizeBody = func(content string) string {
+			newcontent := regexp.MustCompile(`"api_key":"[a-zA-Z0-9_\-\:]+"`).
+				ReplaceAll([]byte(content), []byte(`"api_key":"REDACTED"`))
+			newcontent = regexp.MustCompile(`"secret_token":{"type":"text","value":"[a-zA-Z0-9\-_\:]+"}`).
+				ReplaceAll(newcontent, []byte(`"secret_token":{"type":"text","value":"REDACTED"}`))
+			newcontent = regexp.MustCompile(`"secret_token":"[a-zA-Z0-9\-_\:]+"`).
+				ReplaceAll(newcontent, []byte(`"secret_token":"REDACTED"`))
+			return string(newcontent)
+		}
+
+		i.Request.Body = sanitizeBody(i.Request.Body)
+		i.Response.Body = sanitizeBody(i.Response.Body)
+
+		return nil
+	})
+
+	hc = &http.Client{
+		Transport: rec,
+		Timeout:   10 * time.Second,
+	}
+
+	return rec, hc
+}
+
+func TestClientNew(t *testing.T) {
+	kibanaURL := "http://example.com"
+	apikey := "apikey"
+	c := kbclient.New(kibanaURL, apikey)
+	rec, httpc := getHttpClient(t)
+	defer rec.Stop()
+	c.Client = *httpc
+
+	req, err := http.NewRequest(http.MethodGet, "https://example.com/", nil)
```

Comment:
does it really reach out to example.com? Can we use a mock http server instead?

Reply:
Using `httptest` is problematic because the port changes at each run. A dedicated mock implementation may work, but I don't think this test has enough value to spend effort on it. I removed it.

Reply:
addressed in [416dc12](https://github.com/elastic/apm-server/pull/15818/commits/416dc12c22e5ddcadc66a19347c47c66221a8393)